api/src/lib/interface.ts

```diff
@@ -434,7 +434,7 @@ export interface UpdateReferencePath {
 }
 
 export interface RemoveAllReferencesPath {
-    publicationId: string;
+    id: string;
```

Comment:
I think "publicationId" was a better naming here since we use this interface in the references controller and the publication is not the main resource of this endpoint.
For example: 
- "id" param would mean a Reference ID in the references controller
and
- "id" param would mean a Publication ID in the publications controller

Reply:
Good point, although I decided to change this to `id` as in the severless.yml file the route is `/publications/{id}/reference` (similar to the other routes eg. createReference/ updateReference etc). The interfaces file for the `removeAll` endpoint was `publicationId`, so the endpoint was incorrectly letting us add references to a live publication, as well as letting us add references to publications that don't exist. 

So to change this name to `publicationId` I'd need to change the route in the serverless.yml file to `/publications/{publicationId}/reference` which would be inconsistent with all of the other endpoints, and arguably not RESTful as it's implicit in the route that this is the id of the publication because it comes after the /publication/.

Hopefully that makes sense!

Agree with the interfaces naming, I'll change that now!